WebAPI/Controllers/ContactController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using FluentValidation;
+using Common.Models;
+using Common.Validators;
+using Practice.Services;
+
+namespace WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ContactController : ControllerBase
+    {
+        private readonly IUserService _userService;
+        private readonly IValidator<Contact> _contactValidator;
+        private readonly ILogger<ContactController> _logger;
+
+        public ContactController(
+            IUserService userService, 
+            IValidator<Contact> contactValidator,
+            ILogger<ContactController> logger)
+        {
+            _userService = userService;
+            _contactValidator = contactValidator;
+            this._logger = logger;
+        }
+
+        [HttpGet("{username}")]
+        public ActionResult<Contact?> GetContact(string username)
+        {
+            return Ok(_userService.FindContactByUsername(username));
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<Contact?>> CreateContact(Contact contact)
```

Comment:
Practice: What happens if someone have this API endpoint and can add/update the user's contact without doing a login? In this case we should implement a protection for this endpoint, please research more about API authentication.